src/lib/hooks/useDownloadAsset.ts

```diff
@@ -0,0 +1,26 @@
+import { useCallback } from 'react';
+import { downloadDomainAsset, getDomainAsset } from '../../lib/helpers';
+
+interface UseDownloadAssetReturn {
+	downloadAsset: (assetUrl: string) => Promise<void>;
+}
+
+export const useDownloadAsset = (): UseDownloadAssetReturn => {
+	const downloadAsset = useCallback(async (assetUrl: string) => {
+		try {
+			const asset = await getDomainAsset(assetUrl);
+
+			if (asset) {
+				await downloadDomainAsset(asset);
+			} else {
+				console.error(`No asset found for URL: ${assetUrl}`);
+			}
+		} catch (e: any) {
+			console.error(`Error downloading asset: ${e.message}`);
+		}
+	}, []);
+
+	return {
+		downloadAsset,
+	};
+};
```

Comment:
nit: I think this should be a utility function, rather than a hook, since it doesn't tie into any component state/lifecycle.